app/models/push_notification_target.rb

```diff
@@ -0,0 +1,113 @@
+class PushNotificationTarget < ApplicationRecord
+  resourcify
+
+  FOREM_BUNDLE = "com.forem.app".freeze
+  SUPPORTED_PLATFORMS = [Device::ANDROID, Device::IOS].freeze
+  FOREM_APP_PLATFORMS = [Device::IOS].freeze
+
+  validates :app_bundle, presence: true
+  validates :platform, presence: true
+
+  # Clear Redis-backed model to ensure it will be recreated with updated values
+  after_update :clear_rpush_app
+
+  def forem_app?
+    app_bundle == FOREM_BUNDLE
+  end
+
+  def active?
+    # TRUE if it's marked as `active` in the DB && it has credentials available
+    active && auth_credentials?
+  end
+
+  def auth_credentials?
+    auth_credentials.present?
+  end
+
+  def auth_credentials
+    if app_bundle == FOREM_BUNDLE && platform == Device::IOS
+      ApplicationConfig["RPUSH_IOS_PEM"]
+    else
+      auth_key
+    end
+  end
+
+  def recreate_ios_app!
+    app = Rpush::Apns2::App.new
+    app.name = app_bundle
+    app.certificate = auth_credentials.to_s.gsub("\\n", "\n")
+    app.environment = Rails.env.production? ? "production" : "development"
+    app.password = ""
+    app.bundle_id = app_bundle
+    app.connections = 1
+    app.save!
+    app
+  end
+
+  def recreate_android_app!
+    app = Rpush::Gcm::App.new
+    app.name = app_bundle
+    app.auth_key = auth_credentials.to_s
+    app.connections = 1
+    app.save!
+    app
+  end
```

Comment:
I wonder if this should be part of the model or moved into a `PushNotificationTargets::CreateIosApp` sort of service